### PR TITLE
Serve Doc: Quickstart

### PR DESCRIPTION
**Pull request by simon-mo**
Adds quickstart doc for ray serve. Pages in "Learn more" will be posted in follow up PR

### CODE REVIEW

File: python/ray/serve/examples/doc/quickstart_function.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+serve.init()
+
+
+@serve.route("/hello")
+def echo(flask_request):
+    return "hello " + flask_request.args.get("name", "serve!")
+
+
+requests.get("http://127.0.0.1:8000/hello").text
+# > "hello serve!"
```
Comment (richardliaw) [inline python/ray/serve/examples/doc/quickstart_function.py:12]: please run this in CI?

File: doc/source/serve/quickstart.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    :align: center
+
+Serve is a serving library built on top of ray. It is easy-to-use and flexible.
+
+- Serve allows you to write *stateless* logic in serve functions and *stateful* ones in serve actors.
+- Serve have *Flask-inspired API*. Your directly work with a Flask request object on each request.
+- Serve can grow as your project grow, it can *scale to a cluster* without changing any lines of code.
+- *Everything is in Python*. Instead of writing configuration YAML, the language to configure the serving system is the same as the language to write the application logic.
+- *Batching* and *SLO aware scheduling* are builtin features. Serve can automatically batch queries and reorder them to meet SLO constraint.
+- Deploy to any cloud (or Kubernetes) with Ray autoscaler.
```
Comment (richardliaw) [inline doc/source/serve/quickstart.rst:14]: Grammar

File: python/ray/serve/examples/doc/quickstart_function.py (hunk 1 above)
Comment (simon-mo) [reply python/ray/serve/examples/doc/quickstart_function.py:13]: Added to CI

File: doc/source/index.rst [hunk 3]
```diff
@@ -17,12 +17,14 @@ Ray is packaged with the following libraries for accelerating machine learning w
 - `Tune`_: Scalable Hyperparameter Tuning
 - `RLlib`_: Scalable Reinforcement Learning
 - `RaySGD`_: Distributed Training Wrappers
+- `Ray.Serve`_: Scalable and Programmable Serving
```
Comment (richardliaw) [inline doc/source/index.rst:20]: rayserve? better seo maybe?

File: doc/source/serve/quickstart.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
+Ray Serve: Scalable and Programmable Serving
+============================================
+
+.. image:: logo.svg
+    :align: center
+
+Serve is a serving library built on top of ray. It is easy-to-use and flexible.
+
+- Serve allows you to write *stateless* logic in serve functions and *stateful* logic in serve actors.
```
Comment (richardliaw) [inline doc/source/serve/quickstart.rst:9]: try not to forward reference (don't mention concepts that I don't know about) - for example, what is a serve function?

File: doc/source/serve/quickstart.rst [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+- Serve has *Flask-inspired API*. You directly work with a Flask request object on each request.
```
Comment (richardliaw) [inline doc/source/serve/quickstart.rst:10]: I think the second sentence is a bit confusing.

File: doc/source/serve/quickstart.rst [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+- Serve grows as your project grows, it can *scale to a cluster* without changing any lines of code.
```
Comment (richardliaw) [inline doc/source/serve/quickstart.rst:11]: I'm not sure what this means - if my git repo gets bigger serve gets bigger?

**Review by richardliaw — APPROVED** [review]
Address comments; otherwise lgtm

File: doc/source/serve/quickstart.rst [hunk 7]
```diff
@@ -0,0 +1,45 @@
+Ray Serve: Scalable and Programmable Serving
+============================================
+
+.. image:: logo.svg
```
Comment (richardliaw) [inline doc/source/serve/quickstart.rst:4]: logo doesn't show up for me..

File: doc/source/serve/quickstart.rst [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    :align: center
+
+Serve is a serving library built on top of ray. It is easy-to-use and flexible.
+
+- Serve allows you to write *stateless* logic in serve functions and *stateful* logic in serve actors.
+- Serve has *Flask-inspired API*. You directly work with a Flask request object on each request.
+- Serve grows as your project grows, it can *scale to a cluster* without changing any lines of code.
+- *Everything is in Python*. Instead of writing configuration YAML, the language to configure the serving system is the same as the language to write the application logic.
+- *Batching* and *SLO aware scheduling* are built-in features. Serve can automatically batch queries and reorder them to meet SLO target.
+- Deploy to any cloud (or Kubernetes) with Ray Autoscaler.
```
Comment (richardliaw) [inline doc/source/serve/quickstart.rst:14]: Don't use italics; use bold (** asdfadfa **). It's much easier to read.

File: doc/source/serve/quickstart.rst [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+.. image:: logo.svg
+    :align: center
+
+Serve is a serving library built on top of ray. It is easy-to-use and flexible.
+
+- Serve allows you to write *stateless* logic in serve functions and *stateful* logic in serve actors.
+- Serve has *Flask-inspired API*. You directly work with a Flask request object on each request.
+- Serve grows as your project grows, it can *scale to a cluster* without changing any lines of code.
+- *Everything is in Python*. Instead of writing configuration YAML, the language to configure the serving system is the same as the language to write the application logic.
+- *Batching* and *SLO aware scheduling* are built-in features. Serve can automatically batch queries and reorder them to meet SLO target.
```
Comment (richardliaw) [inline doc/source/serve/quickstart.rst:13]:
```suggestion
- *Batching* and *SLO aware scheduling* are built-in features. Serve can automatically batch queries and reorder them to meet SLO targets.
```

**Comment by anabranch** [comment]
Will review tomorrow AM!

**Pull request merged**